pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -743,3 +743,19 @@ func deleteJob(cj *batchv1.CronJob, job *batchv1.Job, jc jobControlInterface, re
 func getRef(object runtime.Object) (*corev1.ObjectReference, error) {
 	return ref.GetReference(scheme.Scheme, object)
 }
+
+func formatSchedule(cj *batchv1.CronJob, recorder record.EventRecorder) string {
+	if len(cj.Spec.TimeZone) > 0 {
```

Comment:
Instead of checking string length, we may want to check if the field is nil?
"" is a valid value anyway.